### ISSUE TITLE
Reimplement checking for un-preprocessed files in a cross-platform way

### ISSUE THREAD

**Issue by JustOff** [MoonchildProductions/UXP#1497]
Check for unpreprocessed files on Windows was [broken](https://github.com/MoonchildProductions/UXP/commit/e51afbcc2fe7360bbcf5654f6e31752c48098ca0) when adding Solaris support in #1262.

As a result, unprocessed files on Windows are not reported, which can lead to errors.

**Comment by JustOff** [comment]
@athenian200, brace expansion doesn't work on Windows, so you have to find another general solution or provide a specific one for Solaris only.

**Comment by mattatobin** [comment]
How about remove that garbage and just not screw up?

**Comment by JustOff** [comment]
How about not calling a useful check garbage?

**Comment by mattatobin** [comment]
How about not assuming what you consider useful is actually useful. Just slows down the process.

**Comment by JustOff** [comment]
Let me remind you that earlier we [agreed](https://github.com/MoonchildProductions/UXP/issues/428#issuecomment-393859246) that it was useful. As for the time, then, IMHO, an extra 10-15 seconds is not worth mentioning.

As for the fact that no one noticed, we are just lucky that this did not lead to trouble, as it was already more than once. And I absolutely don't understand why you suddenly decided to resist to bring this instrument back to life.

**Comment by mattatobin** [comment]
It was a stressful time then and that issue doesn't have the full story of why I decided to stop obstructing it at that time.

tl;dr that was then this is now and your understanding is not required.

But sure, we can ask @wolfbeast to flip a coin.

**Comment by JustOff** [comment]
I hope you have not forgotten how a small typo can break everything? So, I believe that it's simply unreasonable to refuse an almost free additional automatic check.

> your understanding is not required

Please don't start this again.

**Comment by wolfbeast** [comment]
Right. Stop your bickering.

First off, what was put in is pretty much a dirty hack - while having some use, "just in case" it's also not essential if people actually build and test their changes because it would come out in the wash soon enough if preprocessor directives are in an unprocessed file since it _will_ break.

Secondly, JustOff, you indeed do not have the entire image of why this wasn't obstructed further, so please don't pretend you do. Tobin and I discuss a hell of a lot of things about the project outside of public space in IM that you aren't privy to.

Instead of unloading this on Athenian200, how about you come up with a solution for your original code that works in all operating systems we are currently targeting? If you can't or won't, then we can remove this since it's not working in its current incarnation.

**Comment by JustOff** [comment]
> what was put in is pretty much a dirty hack

You can call it anything you like, but you certainly know that an unprocessed directive can sometimes lead to an error only under certain conditions, which may not be revealed during testing. And this check was added precisely after such an incident and successfully helped to avoid similar problems until it was broken.

> you indeed do not have the entire image of why this wasn't obstructed further, so please don't pretend you do

I don't pretend to anything, except for mutual respect in a public discussion. This is exactly what I asked for.

> Instead of unloading this on Athenian200, how about you come up with a solution for your original code that works in all operating systems we are currently targeting, including Solaris? If you can't or won't, then we can remove this since it's not working in its current incarnation.

Sorry, but this goes against all logic. I could try to help Athenian200 resolve the issue caused by his changes, which made the check incompatible with one of the primary UXP platforms, but obviously I can't do it instead of him, since he is the only developer who uses Solaris.

It seems to me this is just an ordinary case in collaborative development, which does not have any conflict from the very beginning, and I don't understand why the discussion was so escalated.

**Comment by wolfbeast** [comment]
If you want to talk about mutual respect then you can also talk about working _with_ athenian200 to come to a solution instead of just saying "you can't use this this way, fix it" when it was _your_ original code -- meaning you have some responsibility to follow up on issues with it. or don't you understand the concept of owning code you contribute?

Since you're instead treating this now as a "one-shot contribution" and don't want to take responsibility for your code, I'll just back it all out. If there's still interest in such a check (despite it being broken having gone completely unnoticed for months), either or both of you can come up with a solution that works on all supported platforms.

**Comment by JustOff** [comment]
All the assumptions you made above about me and my attitude to the project and other participants are unfair and simply incorrect. I always had and have responsibility for my contribution and always promptly participated in solving any problems if they arise. This has been proven by time and I am very sorry to see your sudden change in attitude that occurred for a completely incomprehensible reason. I understand now is a difficult time, but it would be ridiculous if we would poison relations simply because of this.

**Comment by mattatobin** [comment]
Well you could just found a solution on your own or asked to work with @athenian200 from the start instead of demanding he and he alone fix this. I actually could solve this in 10 minutes or so with testing but I don't actually want to.

Thing is, for the past year, actually after I gave you my 2014 datastore your participation has been sorely lacking except for language packs and some of your extensions that is. You only seem to popup when something of yours is busted, blocked, or threatened. Being evasive and sketchy and all. So where were you when the `Solaris` stuff was landing when your little contribution was touched, hmm?

You completely disrespected @athenian200 and you have never treated me with much of any despite my attempts to reconcile with you. That has been noticed by others btw.

I appreciate that you keep the language packs up and going don't get me wrong on that but there is some kind of expressive barrier on your part that I can't seem to work out and it is very aggravating not only for my self but for some others.

**Comment by JustOff** [comment]
> Well you could have just found a solution on your own or asked to work with @athenian200 from the start instead of demanding he and he alone fix this.

Where, for goodness' sake, did you see that I demanded that he and he alone fix everything? I just pointed out what exactly in his code does not work under Windows. This could be the beginning of a normal discussion, as is usually the case, if you had not interfered with your dismissive commentary.

> Thing is, for the past year [..] your participation has been sorely lacking except for language packs and some of your extensions that is. You only seem to popup when something of yours is busted, blocked, or threatened. Being evasive and sketchy and all.

This is not true, however it has nothing to do with this issue. I don't believe you have been accumulating your discontent for a whole year, only to use it at an opportunity.

> So where were you when the Solaris stuff was landing when your little contribution was touched, hmm?

I never promised to track every single change to the entire codebase, and aren't the contributor and reviewer responsible for making sure the code in which the changes are made remains working properly? But even when a flaw was discovered, did I make a drama out of it?

> You completely disrespected @athenian200, someone who is under my direct mentorship

Once again, I discovered the problem and started the discussion by creating an issue, without a single bad word or accusation. You simply invented the rest, having inflated the conflict out of the blue. And of course, if @athenian200 says that I really offended him with something, I am ready to apologize. But this does not require intermediaries, and he has not yet said a word.

> and you have never treated me with much of any despite my attempts to reconcile with you. That has been noticed by others btw.

It reminds me of a quote from one famous movie: "Now you come to me and say 'Don Corleone, give me justice'. But you don't even ask with respect, you don't offer friendship, you don't even think to call me Godfather". I always appeal to you with due respect, but, excuse me, you are not the Godfather to receive my obeisance.

> there is some kind of expressive barrier on your part that I can't seem to work out and it is very aggravating not only for my self but for some others.

Sorry, but this barrier is entirely in your mind because with everyone else around here I have always been able to resolve issues without conflict. And you could probably notice that for my part I constantly make special efforts to avoid situations when our public interaction could negatively affect the project. Maybe you should just try doing the same?

**Comment by wolfbeast** [comment]
> Where, for goodness' sake, did you see that I demanded that he and he alone fix everything?

Here:

> @athenian200, brace expansion doesn't work on Windows, so you have to find another general solution or provide a specific one for Solaris only.

which most definitely reads as a command from you to him "you broke something, and you have to fix it". No ambiguity there.

>But even when a flaw was discovered, did I make a drama out of it?

You just did, right here, in this issue.

Also, you can count me among "some others" that Tobin is alluding to. I have to agree you've been elusive for the most part, yet needing immediate attention (and will push for it) when something that directly influences you is not working as it should.
I also understand you are protective of your work and unhappy that someone else's contribution broke it (which they can't be faulted for, given the circumstances) but if someone doesn't have the knowledge or capacity/environment to address this on their own in code that they didn't write but you did, then you **can't** just tell them to go fix it. and certainly not make a long drama issue out of it.

> Sorry, but this barrier is entirely in your mind

Then it has to be in mine too. And possibly a few others' -- draw your own conclusions from that.

**Comment by JustOff** [comment]
> which most definitely reads as a command from you to him "you broke something, and you have to fix it".

Just in the previous comment, I already explained what exactly I had in mind when referring to @athenian200.

> you can't just tell them to go fix it

And I didn't mean it. Please open your mind and hear my arguments. I try to do everything possible to resolve this causeless quarrel, but it requires good will from all parties.

> Then it has to be in mine too.

I'm really sorry to hear that.

<hr>

Despite the fact that this issue has already been closed, let me nevertheless return to its essence.

> what IS needed is for people to take more CARE in what they submit and more CARE in what is reviewed and accepted

I would not start this discussion again if today I hadn't accidentally discovered [unprocessed directives](https://github.com/MoonchildProductions/UXP/blob/6a3907dae474c7e1f31df13a26605278d3500a0d/toolkit/modules/addons/WebRequest.jsm#L843-L845) in `WebRequest.jsm` that were added there [a month ago](https://github.com/MoonchildProductions/UXP/commit/6a3907dae474c7e1f31df13a26605278d3500a0d#diff-b6c92703808b0843d15b077e7f505dca) and then got into the latest release, although this could have been prevented by using automatic check.

**Comment by mattatobin** [comment]
Taking this so the crying stops.
